### ISSUE TITLE
My site: VoiceOver looses track of selected item when triggering navigation a section using the buttons in the header 

### ISSUE THREAD

**Issue by ctarda** [wordpress-mobile/WordPress-iOS#14999]
When navigating into any of the subsections of My Site using the button in the My Site header (those within the red rectangle in the screenshot) and navigating back to My Site, Voice Over resets the cursor to the top of the view.

However, if I navigate to any of those subpages using the table view rows, everything works as expected

Screenshot highlighting the buttons inside a red rectangle: 
<Img src="https://user-images.githubusercontent.com/2722505/94389023-3d8d2000-0181-11eb-8ad9-80b23a048d0c.jpeg" width="350" alt="Screenshot highlighting the problematic buttons inside a red rectangle"/>

### Expected behavior
When I navigate into any of the sections offered in the My Site header (Stats, Pages, Posts or Media) and then I navigate back to My Site, I expect the VoiceOver cursor to remain in the button I tapped. My expectation is:

1. Tap Stats
2. Tap Back
3. Stats should be active

### Actual behavior
<Img src="https://user-images.githubusercontent.com/2722505/94389690-1df6f700-0183-11eb-8fdf-61933cf84ac3.gif" width="350" alt="Recording of the behaviour observed. When navigating to stats, and then back to My Site, the navigation cursor jumps to the navigation bar" />
1. Tap Stats
2. Tap Back
3. Stats is not active. Back is.

### Steps to reproduce the behavior

1. Activate VoiceOver
2. Navigate to My Site
3. Select the Stats button in the header, either by double tapping on it or by swapping right until VoiceOver reaches it
4. From stats, navigate back to My Site, either shipping back until reaching the Back button and double tapping it, or by selecting the Back button directly and double tapping it.
5. VoiceOver's selection cursor jumps to My Site's back button (top left of the navigation bar)

##### Tested on [device], iOS [version], WPiOS [version]
Tested on iPhone 6s+, iOS 13.7, WPiOS 15.7, es_ES
Tested on iPhone 11 Pro Max, iOS 14.0.1, WPiOS 15.7, en_CA

**Comment by dangermattic** [comment]
Thanks for reporting! 👍
